src/GraphQL/Validation/Rules/VariablesAreInputTypes.cs

```diff
@@ -21,14 +21,22 @@ public Task<INodeVisitor> ValidateAsync(ValidationContext context)
             {
                 _.Match<VariableDefinition>(varDef =>
                 {
-                    var type = varDef.Type.GraphTypeFromType(context.Schema);
+                    var type = GetNamedGraphTypeFromType(varDef.Type, context.Schema);
 
                     if (!type.IsInputType())
```

Comment:
`IsInputType()` unwraps `type` so there is no reason to call expensive `GraphTypeFromType` here